cerberus/errors.py

```diff
@@ -525,89 +535,76 @@ def insert_error(self, path, node):
+                self._insert_error(child_error.document_path,
+                                   self._format_message(child_error.field, child_error))
 
-    def insert_logic_error(self, error):
+    def _insert_logic_error(self, error):
         field = error.field
-        self.insert_error(error.document_path,
-                          self.format_message(field, error))
-
-        for child_errors in error.definitions_errors.values():
-            if not child_errors:
-                continue
+        self._insert_error(error.document_path,
+                           self._format_message(field, error))
 
-            for child_error in child_errors:
+        for definition_errors in error.definitions_errors.values():
+            for child_error in definition_errors:
                 if child_error.is_logic_error:
-                    self.insert_logic_error(child_error)
+                    self._insert_logic_error(child_error)
                 elif child_error.is_group_error:
-                    self.insert_group_error(child_error)
+                    self._insert_group_error(child_error)
                 else:
-                    self.insert_error(child_error.document_path,
-                                      self.format_message(field, child_error))
+                    self._insert_error(child_error.document_path,
+                                       self._format_message(field, child_error))
+
+    def _purge_empty_dicts(self, error_list):
+        subtree = error_list[-1]
+        if not error_list[-1]:
+            error_list.pop()
+        else:
+            for key in subtree:
+                self._purge_empty_dicts(subtree[key])
 
-    def rewrite_error_path(self, error, offset=0):
+    def _rewrite_error_path(self, error, offset=0):
         """
         Recursively rewrites the error path to correctly represent logic errors
         """
         if error.is_logic_error:
-            self.rewrite_logic_error_path(error, offset)
+            self._rewrite_logic_error_path(error, offset)
         elif error.is_group_error:
-            self.rewrite_group_error_path(error, offset)
+            self._rewrite_group_error_path(error, offset)
 
-    def rewrite_group_error_path(self, error, offset=0):
+    def _rewrite_group_error_path(self, error, offset=0):
         child_start = len(error.document_path) - offset
 
         for child_error in error.child_errors:
             relative_path = child_error.document_path[child_start:]
             child_error.document_path = error.document_path + relative_path
 
-            self.rewrite_error_path(child_error, offset)
+            self._rewrite_error_path(child_error, offset)
 
-    def rewrite_logic_error_path(self, error, offset=0):
-        error.document_path = error.document_path + (error.rule, )
-        child_start = len(error.document_path) - offset - 1
+    def _rewrite_logic_error_path(self, error, offset=0):
+        error.document_path = error.document_path
```

Comment:
😄

Reply:
:-D looks like a missing feature for `flake8`.